### PR TITLE
Add description on how to use GDB

### PR DESCRIPTION
**Pull request by mchack-work**
Still to do: Figure out how to install the right GDB on Ubuntu. Do we point people to https://github.com/riscv-collab/riscv-gnu-toolchain/releases/ ?

### CODE REVIEW

File: hugo/content/tools/_index.md [hunk 1]
```diff
@@ -245,7 +245,23 @@ You can also use the QEMU monitor for debugging, for example, `info
 registers`, or run QEMU with `-d in_asm` or `-d trace:riscv_trap` for
 tracing.
 
-TODO Give examples on how to use gdb with qemu.
+If you run QEMU with `-s` it provides the GDB protocol on
+`localhost:1234` (default). If you run with `-S` qemu doesn't start
```
Comment (quite) [inline hugo/content/tools/_index.md:249]: Is `-S` to be used (optionally) in addition to `-s`?

File: hugo/content/tools/_index.md [hunk 2]
```diff
@@ -245,7 +245,23 @@ You can also use the QEMU monitor for debugging, for example, `info
...
+automatically and you can control it entirely from GDB.
+
+TODO: Install GDB with support for rv32.
+
+To attach GDB do something like:
+
+```
+riscv32-elf-gdb firmware.elf --tty=/dev/pts/10 \
```
Comment (quite) [inline hugo/content/tools/_index.md:257]: What is this tty to be passed here?

File: hugo/content/tools/_index.md [hunk 3]
```diff
@@ -245,7 +245,23 @@ You can also use the QEMU monitor for debugging, for example, `info
 registers`, or run QEMU with `-d in_asm` or `-d trace:riscv_trap` for
 tracing.
 
-TODO Give examples on how to use gdb with qemu.
+If you run QEMU with `-s` it provides the GDB protocol on
+`localhost:1234` (default). If you run with `-S` qemu doesn't start
+automatically and you can control it entirely from GDB.
+
+TODO: Install GDB with support for rv32.
```
Comment (quite) [inline hugo/content/tools/_index.md:252]: I didn't find anything but the riscv-collab repo either, after a quick search

**Comment by quite** [comment]
Looks good.

**Comment by niels-moller** [comment]
Would be nice if docs said whether or not the debian-package gdb-multiarch (which has some riscv support) is expected to work. Maybe that's available on ubuntu as well?

**Pull request merged**